src/mame/video/hng64_sprite.hxx

```diff
@@ -56,19 +58,63 @@ void hng64_state::draw_sprites(screen_device &screen, bitmap_rgb32 &bitmap, cons
 	osd_printf_debug("\n");
 #endif
 
-	while( source<finish )
+	while(source < finish)
 	{
+		if (source[4]&0x04000000) // disable bit, ss64 rankings ?
+		{
+			source += 8;
+		}
+		else
+		{
+			if (m_spriteregs[0] & 0x01000000)
+		// This flips between ingame and other screens for roadedge, where the sprites which are filtered definitely needs to change and the game explicitly swaps the values in the sprite list at the same time.
+		// m_spriteregs[2] could also play a part as it also flips between 0x00000000 and 0x000fffff at the same time
+		// Samsho games also set the upper 3 bits which could be related, samsho games still have some unwanted sprites (but also use the other 'sprite clear' mechanism)
+		// Could also be draw order related, check if it inverts the z value?
+			{
```

Comment:
Please don’t put anything between the control statement and the start of the controlled compound statement – it’s confusing, especially when the comment isn’t indented to match the code like this.